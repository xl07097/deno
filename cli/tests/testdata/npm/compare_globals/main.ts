/// <reference types="npm:@types/node" />

import * as globals from "npm:@denotest/globals";
console.log(globals.global === globals.globalThis);
console.log(globals.process.execArgv);

type AssertTrue<T extends true> = never;
type _TestNoProcessGlobal = AssertTrue<
  typeof globalThis extends { process: any } ? false : true
>;
type _TestHasNodeJsGlobal = NodeJS.Architecture;

const controller = new AbortController();
controller.abort("reason"); // in the NodeJS declaration it doesn't have a reason

// Super edge case where some Node code deletes a global where the
// Node code has its own global and the Deno code has the same global,
// but it's different. Basically if some Node code deletes
// one of these globals then we don't want it to suddenly inherit
// the Deno global.
globals.withNodeGlobalThis((nodeGlobalThis: any) => {
  (globalThis as any).setTimeout = 5;
  console.log(setTimeout);
  delete nodeGlobalThis["setTimeout"];
  console.log(nodeGlobalThis["setTimeout"]); // should be undefined
  console.log(globalThis["setTimeout"]); // should be undefined
});
