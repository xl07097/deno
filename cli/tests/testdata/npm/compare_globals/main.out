Download http://localhost:4545/npm/registry/@types/node
Download http://localhost:4545/npm/registry/@denotest/globals
Download http://localhost:4545/npm/registry/@denotest/globals/1.0.0.tgz
Download http://localhost:4545/npm/registry/@types/node/node-18.8.2.tgz
Check file:///[WILDCARD]/npm/compare_globals/main.ts
true
[]
5
undefined
undefined
